src/workerd/api/container.h

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2025 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+// APIs that an Actor (Durable Object) uses to access its own state.
+//
+// See actor.h for APIs used by other Workers to talk to Actors.
+
+#include <workerd/io/container.capnp.h>
+#include <workerd/io/io-own.h>
+#include <workerd/jsg/jsg.h>
+
+namespace workerd::api {
+
+class Fetcher;
+
+// Implements the `ctx.container` API for durable-object-attached containers. This API allows
+// the DO to supervise the attached container (lightweight virtual machine), including starting,
+// stopping, monitoring, making requests to the container, intercepting outgoing network requests,
+// etc.
+class Container: public jsg::Object {
+ public:
+  Container(rpc::Container::Client rpcClient);
+
+  struct StartupOptions {
+    jsg::Optional<kj::Array<kj::String>> entrypoint;
+    bool enableInternet = false;
+
+    // TODO(containers): stdio interception
+
+    JSG_STRUCT(entrypoint, enableInternet);
+  };
+
+  bool getRunning() {
+    return running;
+  }
+
+  // Methods correspond closely to the RPC interface in `container.capnp`.
+  void start(jsg::Lock& js, jsg::Optional<StartupOptions> options);
+  jsg::Promise<void> monitor(jsg::Lock& js);
+  jsg::Promise<void> destroy(jsg::Lock& js, jsg::Optional<jsg::Value> error);
+  void signal(jsg::Lock& js, int signo);
+  jsg::Ref<Fetcher> getTcpPort(jsg::Lock& js, int port);
```

Comment:
Nit: The naming of this is a bit weird and could use a code comment about what is intended here. The `get{whatever}` pattern is most typically seen for property getters but this is a method that takes a port? Just feels a bit awkward.

Reply:
I think there's plenty of precedent for functions that start with `get` but are not simple property getters. E.g. `DO_NAMESPACE.get(id)`.

Per the comment above this code block, all these methods directly correspond to methods on the capnp interface where they are documented in detail. I didn't want to copy all the comments and have to keep them in sync.

Reply:
That said if you have a better naming idea I'm totally open to suggestions. This interface won't be public for a while.

Reply:
No renaming suggestions but a code comment explaining that would be worthwhile for future folks coming into this code.